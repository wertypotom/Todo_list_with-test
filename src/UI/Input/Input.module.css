.container {
  background-color: white;
  display: flex;
  align-items: center;
  padding: 0 10px;

  width: 100%;
  border-bottom: 2px solid var(--secondary-color);
  box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.1);
  border-radius: var(--radius);
}

input[type='text'] {
  width: 100%;
  font-size: 20px;
  letter-spacing: 2px;
  font-weight: 100;

  padding: 20px;
  outline: none;
  border: none;

  font-weight: lighter;
}

input::placeholder {
  font-weight: 100;
  color: var(--secondary-color-accent);
  font-style: italic;
}

.icon {
  width: 30px;
  cursor: pointer;
}
