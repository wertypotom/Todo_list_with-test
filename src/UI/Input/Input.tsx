import React from 'react';
import styles from './Input.module.css';
import { ArrowDown } from '../Icons/ArrowDown';

interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
  onFormSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
  onArrowClick?: () => void;
}

const Input = ({ onFormSubmit, onArrowClick, ...rest }: Props) => {
  return (
    <form onSubmit={onFormSubmit} className={styles.container}>
      <ArrowDown
        className={styles.icon}
        onClick={onArrowClick}
        data-testid='arrow-down'
      />
      <input autoFocus type='text' {...rest} />
    </form>
  );
};

export default Input;
