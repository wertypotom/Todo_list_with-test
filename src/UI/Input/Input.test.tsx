import { fireEvent, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Input from './Input';

describe('UI Input', () => {
  it('renders correctly', () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });

  it('correctly inputs data', async () => {
    const onChange = vi.fn();
    const { container } = render(<Input onChange={onChange} />);

    const input = container.querySelector('input')!;

    await userEvent.type(input, 'Hello');

    expect(onChange).toHaveBeenCalledTimes(5);
    expect(input).toHaveValue('Hello');
  });

  it('calls onFormSubmit when form is submitted', async () => {
    const onFormSubmitMock = vi.fn();
    const { container } = render(<Input onFormSubmit={onFormSubmitMock} />);

    const form = container.querySelector('form') as HTMLElement;

    fireEvent.submit(form);
    expect(onFormSubmitMock).toHaveBeenCalledTimes(1);
  });

  it('calls onArrowClick when arrow icon is clicked', async () => {
    const onArrowClickMock = vi.fn();
    const { getByTestId } = render(<Input onArrowClick={onArrowClickMock} />);

    const arrowIcon = getByTestId('arrow-down');

    await userEvent.click(arrowIcon);
    expect(onArrowClickMock.mock.calls.length).toBe(1);
  });
});
