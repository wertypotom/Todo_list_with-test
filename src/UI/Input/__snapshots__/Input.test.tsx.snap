// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UI Input > renders correctly 1`] = `
<div>
  <form
    class="_container_db8317"
  >
    <svg
      class="_icon_db8317"
      data-testid="arrow-down"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m12 15.4l-6-6L7.4 8l4.6 4.6L16.6 8L18 9.4l-6 6Z"
        fill="gray"
      />
    </svg>
    <input
      type="text"
    />
  </form>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <form
    class="_container_db8317"
    data-testid="input-form"
  >
    <svg
      class="_icon_db8317"
      data-testid="arrow-down"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m12 15.4l-6-6L7.4 8l4.6 4.6L16.6 8L18 9.4l-6 6Z"
        fill="gray"
      />
    </svg>
    <input
      type="text"
    />
  </form>
</div>
`;
